doc/sphinx/language/core/basic.rst

```diff
@@ -0,0 +1,428 @@
+
+  All characters are meaningful. In particular, identifiers are
+  case-sensitive.  :production:`unicode_letter` non-exhaustively
+  includes Latin, Greek, Gothic, Cyrillic, Arabic, Hebrew, Georgian,
+  Hangul, Hiragana and Katakana characters, CJK ideographs,
+  mathematical letter-like symbols and non-breaking
+  space. :production:`unicode_id_part` non-exhaustively includes
+  symbols for prime letters and subscripts.
+
+Numerals
+  Numerals are sequences of digits with an optional fractional part
+  and exponent, optionally preceded by a minus sign. :n:`@int` is an
+  integer; a numeral without fractional or exponent parts. :n:`@num`
+  is a non-negative integer.  Underscores embedded in the digits are
+  ignored, for example ``1_000_000`` is the same as ``1000000``.
+
+  .. insertprodn numeral digit
+
+  .. prodn::
+     numeral ::= {+ @digit } {? . {+ @digit } } {? {| e | E } {? {| + | - } } {+ @digit } }
+     int ::= {? - } {+ @digit }
+     num ::= {+ @digit }
+     digit ::= 0 .. 9
+
+Strings
+  Strings begin and end with ``"`` (double quote).  Use ``""`` to
+  represent a double quote character within a string.  In the grammar,
+  strings are identified with :production:`string`.
+
+Keywords
+  The following character sequences are reserved keywords that cannot
+  be used as identifiers::
+
+    _ Axiom CoFixpoint Definition Fixpoint Hypothesis IF Parameter Prop
+    SProp Set Theorem Type Variable as at by cofix discriminated else
+    end exists exists2 fix for forall fun if in lazymatch let match
+    multimatch return then using where with
+
+  Note that notations and plugins may define additional keywords.
+
+Other tokens
+  The set of tokens defined at any given time can vary because the
+  :cmd:`Notation` command can define new tokens.  A :cmd:`Require`
+  command may load more notation definitions, while the end of a
+  :cmd:`Section` may remove notations.  Some notations are defined in
+  the standard library (see :ref:`thecoqlibrary`) and are generally
+  loaded automatically at startup time.
+
+  Here are the character sequences that Coq directly defines as tokens
+  without using :cmd:`Notation`::
+
+    ! #[ % & ' ( () (bfs) (dfs) ) * ** + , - ->
+    . .( .. ... / : ::= := :> :>> ; < <+ <- <:
+    <<: <= = => > >-> >= ? @ @{ [ [= ] _
+    `( `{ { {| | |- || }
+
+  When multiple tokens match the beginning of a sequence of
+  characters, the longest matching token is used.  Occasionally you
+  may need to insert spaces to separate tokens.  For example, if ``~``
+  and ``~~`` are both defined as tokens, the inputs ``~ ~`` and ``~~``
+  generate different tokens, whereas if `~~` is not defined, then the
+  two inputs are equivalent.
+
+Basic notions
+-------------
+
+Glossary
+~~~~~~~~
+
+.. glossary::
+
+   term
+
+     Terms are syntactically valid objects constructed from the
+     following grammar.  When they are :term:`well-typed`, they
+     represent mathematical objects.
```

Comment:
You can work with untyped terms in Ltac using `uconstr`.  Or you can fake well-typedness by cheating with `exact_no_check`.  But insofar as there's a map from syntax to semantics (formally, from Coq ASTs to sets or types or classes or points in a type), only the well-typed terms get mapped to something.

Reply:
Right. I didn't want to write just "they represent mathematical objects" because `0 1` is syntactically valid and not well-typed and does not carry any meaning.

Reply:
How about something more like this:

```
Definitions, theorems, proofs and other constructs are represented in |Coq| as *terms*.
Terms are constructed from the following grammar.  Generally, |Coq| works with
terms that are well-typed.
```

We should define open_term (aka uconstr) somewhere.  Could be here or elsewhere.

Reply:
"mathematical objects" would need to be defined.  It's better to avoid introducing this term.

Reply:
OK, I agree that my text was not substantial enough. Will change to look more like what you suggest. Regarding `open_term`, this is really only useful to explain for people who write tactics, so to be left for the Ltac chapter (especially as we will move the Tactic Notation stuff over there).

Reply:
Check what I wrote.